src/components/TopQueries/TopQueries.ts

```diff
@@ -0,0 +1,123 @@
+import {
+    Component,
+    ComponentOptions,
+    IAnalyticsActionCause,
+    IComponentBindings,
+    Initialization,
+    IQuerySuggestRequest,
+    l,
+    IQuerySuggestResponse,
+    IAnalyticsNoMeta,
+} from 'coveo-search-ui';
+import './Strings';
+
+export interface ITopQueriesOptions {
+    /**
+     * The parameters sent to the suggestion query.
+     * The component uses this information to get better suggestions
+     * In most cases, the q attribute should be set to ''
+     */
+    suggestionQueryParams?: IQuerySuggestRequest;
+    /**
+     * The displayed title over the suggestions
+     */
+    title?: string;
+    /**
+     * Specifies the handler called when a suggestion is clicked.
+     *
+     * Default executes a search query using the suggestion
+     */
+    onClick?: (expression: string, component: TopQueries) => void;
+}
+
+/**
+ * The TopQueries component suggests the top searched queries in the specific context and links the search results of the suggestions to the user
+ */
+export class TopQueries extends Component {
+    static ID = 'TopQueries';
+
+    /**
+     * The possible options for the TopQueries component
+     * @componentOptions
+     */
+    static options: ITopQueriesOptions = {
+        suggestionQueryParams: ComponentOptions.buildJsonOption<IQuerySuggestRequest>({ defaultValue: { q: '' } }),
+        title: ComponentOptions.buildStringOption({ defaultValue: l('TopQueries_title') }),
+        onClick: ComponentOptions.buildCustomOption((s) => null, {
```

Comment:
Why do you have a lambda that strictly do nothing ?

Reply:
This argument translates a string option to a specified type. Since that type is a function, we only accept it beeing defined in JS when initializing the component.